spring-integration-file/src/main/java/org/springframework/integration/file/remote/AbstractRemoteFileStreamingMessageSource.java

```diff
@@ -136,11 +157,20 @@ protected Object doReceive() {
 			String remotePath = remotePath(file);
 			Session<?> session = this.remoteFileTemplate.getSession();
 			try {
-				return getMessageBuilderFactory().withPayload(session.readRaw(remotePath))
+				AbstractIntegrationMessageBuilder<InputStream> builder = getMessageBuilderFactory()
+						.withPayload(session.readRaw(remotePath))
 						.setHeader(IntegrationMessageHeaderAccessor.CLOSEABLE_RESOURCE, session)
 						.setHeader(FileHeaders.REMOTE_DIRECTORY, file.getRemoteDirectory())
-						.setHeader(FileHeaders.REMOTE_FILE, file.getFilename())
-						.build();
+						.setHeader(FileHeaders.REMOTE_FILE, file.getFilename());
+				if (getObjectMapper() != null) {
+					try {
+						builder.setHeader(FileHeaders.REMOTE_FILE_INFO, getObjectMapper().toJson(file));
```

Comment:
I wonder why can't we build JSON manually from such a simple `FileInfo` object?..
To be honest I'm not sure that we should serialize to the JSON a target implementation returned by the `getFileInfo()`.